flake: make it easier to use devenv in Nix flake (#150)
cachix/devenv: docs/guides/using-with-flakes-devshell.md

> `devenv` is partly a CLI, but also a way to configure a development shell declaratively. This configuration can also be used without using the `devenv` CLI inside other Nix flakes.

Reviewer: ```suggestion
`devenv` can be used without the `devenv` CLI by integrating into [Nix Flakes](https://www.tweag.io/blog/2020-05-25-flakes/) , if you're more familiar with the Nix language/ecosystem.
```

The target audience for this guide are experienced Nix developers, so I want to make sure the first sentence catches that.

Some usecases for using devenv configuration inside flakes is for projects that want to define other Nix flake features, apart from the development shell.
These include a Nix package for the project, NixOS and home-manager modules related to the project.
Usually you want to use the same lock file for the development shell as well as the Nix package and others, so that everything is based on the same nixpkgs.

A Nix flake includes the inputs from `devenv.yaml` as well as the devenv configuration that you'd usually find in `devenv.nix`. `flake.lock` is the lock file for Nix flakes, the equivalent to `devenv.lock`.

## Getting started

> To quickly set a project up with Nix flakes with devenv configuration, you can make use of `nix flake init`, like:

Reviewer: ```suggestion
To quickly set a project up with Nix flakes, use of `nix flake init`, like:
```

```console
$ nix flake init --template github:cachix/devenv
```

> This will create a `flake.nix` with devnenv configuration, as well as a `.envrc` direnv configuration.

Reviewer: ```suggestion
This will create a `flake.nix` with devenv configuration, as well as a `.envrc` direnv configuration.
```

> You can open the devenv shell using:

Reviewer: ```suggestion
Open the devenv shell using:
```

```console
$ nix develop
```

> This will create a lock file, if it wasn't already, and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.

Reviewer: ```suggestion
This will create a lock file and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.
```

It's not that important to explain all those edge cases :)

In this case you want to define a `flake.nix` file that contains multiple `devShells`. For example:

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
    devenv.url = "github:cachix/devenv";
  };

  outputs = { self, nixpkgs, devenv, ... } @ inputs:
    let
      pkgs = import nixpkgs {
        system = "x86_64-linux";
      };
    in
    {
      devShell.x86_64-linux.projectA = devenv.lib.mkShell {
        inherit inputs pkgs;
        modules = [
          {
            enterShell = ''
              echo this is project A
            '';
          }
        ];
      };
      devShell.x86_64-linux.projectB = devenv.lib.mkShell {
        inherit inputs pkgs;
        modules = [
          {
            enterShell = ''
              echo this is project B
            '';
          }
        ];
      };
    };
}
```

Here you can see that there are 2 shells defined. Each one with a devenv configuration with differently defined `enterShell`.

To enter the shell of `projectA`:

```console
$ nix develop .#projectA
this is project A
(devenv) $ 
```

To enter the shell of `projectB`:

```console
$ nix develop .#projectB
this is project B
(devenv) $ 
```

> ### Projects with external flake

Reviewer: ```suggestion
### Projects with an external flake
```
